### PR TITLE
EPMRPP-34325 React. Launches. Modals. Force finish

### PR DESCRIPTION
**Pull request by AmsterGet**

### CODE REVIEW

File: app/src/pages/inside/launchesPage/launchSuiteGrid/hamburger/hamburgerMenuItem/hamburgerMenuItem.jsx [hunk 1]
```diff
@@ -4,17 +4,23 @@ import styles from './hamburgerMenuItem.scss';
 
 const cx = classNames.bind(styles);
 
-export const HamburgerMenuItem = ({ onClick, text, disabled }) => (
-  <div className={cx('hamburger-menu-item', { disabled })} onClick={disabled ? null : onClick}>
+export const HamburgerMenuItem = ({ onClick, text, title, disabled }) => (
+  <div
+    className={cx('hamburger-menu-item', { disabled })}
+    title={title}
+    onClick={disabled ? null : onClick}
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/launchesPage/launchSuiteGrid/hamburger/hamburgerMenuItem/hamburgerMenuItem.jsx]: !disabled && onClick

File: app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    };
+    const cancelButton = {
+      text: intl.formatMessage(COMMON_LOCALE_KEYS.CANCEL),
+    };
+    return (
+      <ModalLayout
+        title={intl.formatMessage(messages.finishForceLaunchHeader)}
+        okButton={okButton}
+        cancelButton={cancelButton}
+        warningMessage={
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx]: (item.owner !== this.props.userId) && intl.formatMessage(messages.finishForceLaunchWarning)

File: app/src/pages/inside/launchesPage/launchSuiteGrid/hamburger/hamburgerMenuItem/hamburgerMenuItem.jsx (hunk 1 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/launchSuiteGrid/hamburger/hamburgerMenuItem/hamburgerMenuItem.jsx]: fixed

File: app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx (hunk 2 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx]: fixed

File: app/src/pages/inside/common/launchSuiteGrid/hamburger/hamburger.jsx [hunk 3]
```diff
@@ -167,15 +172,26 @@ export class Hamburger extends Component {
             )}
             <HamburgerMenuItem
               text={intl.formatMessage(messages.forceFinish)}
+              title={
+                (!canForceFinishLaunch(
```
Comment (DiscoElevator) [inline app/src/pages/inside/common/launchSuiteGrid/hamburger/hamburger.jsx]: Please make it a class method and rewrite (maybe using `if..else` or temporary constants or both), it's unreadable now.

File: app/src/controllers/launch/actionCreators.js [hunk 4]
```diff
@@ -27,6 +27,16 @@ export const proceedWithValidItemsAction = createProceedWithValidItemsAction(NAM
 
 const MODAL_COMPARE_WIDTH = 900;
 
+export const finishForceLaunchesAction = defineGroupOperation(
```
Comment (DiscoElevator) [inline app/src/controllers/launch/actionCreators.js]: forceFinishLaunchesAction? I think it sounds better

File: app/src/pages/inside/launchesPage/launchesPage.jsx [hunk 5]
```diff
@@ -157,6 +165,15 @@ export class LaunchesPage extends Component {
       data: { onConfirm: () => this.moveToDebug(item.id) },
     });
   };
+
+  finishForceLaunches = (eventData) => {
+    let launches = [];
+    eventData.id ? (launches = [eventData]) : (launches = this.props.selectedLaunches);
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/launchesPage.jsx]: `const launches = eventData && eventData.id ? [eventData] : this.props.selectedLaunches`

File: app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+      fetchFunc: () => {},
+      items: [],
+    },
+    showNotification: () => {},
+  };
+
+  finishAndClose = (closeModal) => {
+    const { items, fetchFunc } = this.props.data;
+    const entities = {};
+    items.forEach((item) => {
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx]:
```
const entities = items.reduce((acc, item) => ({
  ...acc,
  [item.id]: {
    end_time: Date.now(),
    status: STOPPED, // make it a constant and add to the `launchStatuses.js`
  },
}), {});
```

File: app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        entities,
+      },
+    })
+      .then(() => {
+        const successMessage =
+          (items.length > 1 && messages.forceFinishSuccessMultipleMessage) ||
+          messages.forceFinishSuccessMessage;
+        this.props.showNotification({
+          message: this.props.intl.formatMessage(successMessage),
+          type: 'success',
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx]: please use special constants from 'controllers/notification'

File: app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+      })
+      .catch((response) => {
+        const errorMessage =
+          (items.length > 1 && messages.forceFinishFailedMultipleMessage) ||
+          messages.forceFinishFailedMessage;
+        this.props.showNotification({
+          message: this.props.intl.formatMessage(errorMessage, {
+            message: response.message,
+          }),
+          type: 'error',
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx]: please use special constants from 'controllers/notification'

File: app/src/pages/inside/common/launchSuiteGrid/hamburger/hamburger.jsx (hunk 3 above)
Comment (AmsterGet) [reply app/src/pages/inside/common/launchSuiteGrid/hamburger/hamburger.jsx]: Done

File: app/src/controllers/launch/actionCreators.js (hunk 4 above)
Comment (AmsterGet) [reply app/src/controllers/launch/actionCreators.js]: Fixed

File: app/src/pages/inside/launchesPage/launchesPage.jsx (hunk 5 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/launchesPage.jsx]: Done

File: app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx (hunk 6 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx]: Done

File: app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx (hunk 7 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx]: Fixed

File: app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx (hunk 8 above)
Comment (AmsterGet) [reply app/src/pages/inside/launchesPage/modals/launchFinishForceModal/launchFinishForceModal.jsx]: Fixed

**Pull request merged**